spec/interactors/update_password_spec.rb

```diff
@@ -0,0 +1,54 @@
+require "rails_helper"
+
+RSpec.describe UpdatePassword do
+  describe ".call" do
+    let(:user) { create(:confirmed_user) }
+
+    let(:user_params) do
+      { password: "newpass", password_confirmation: "newpass" }
+    end
+
+    let(:salt) { Encryptor.generate_salt }
+
+    context "when successful" do
+      subject do
+        described_class.call(user_params: user_params, user: user)
+      end
+
+      it "succeeds" do
+        is_expected.to be_a_success
+      end
+
+      it "updates password_hash" do
+        expect(subject.user.password_hash).
+          to eq(Encryptor.digest_password(user_params[:password],
+                                          user.password_salt))
+      end
+
+      it "updates password_salt" do
+        orig_salt = user.password_salt
+        described_class.call(user_params: user_params, user: user)
+        new_salt = user.password_salt
+        expect(orig_salt).not_to eq(new_salt)
+      end
+
+      it "sets reset_digest to nil" do
+        expect(subject.user.reset_digest).to be_nil
+      end
+    end
+
+    context "when unsuccessful" do
```

Comment:
@Enriikke perhaps I should be more explicit here? I.e. "when password confirmation does not match password"?